--- walk.rs.as-copied	2026-09-24
+++ walk.rs.fixed	2026-09-24
@@ -2365,7 +2365,7 @@
             }
             let commit_marked_counters: Vec<u64> = members
                 .iter()
-                .filter(|(_, member)| member.is_commit)
+                .filter(|(_, member)| matches!(member.commit_marker, CommitMarker::Present))
                 .map(|(counter, _)| *counter)
                 .collect();
             let commit_marker_misplaced = match commit_marked_counters.as_slice() {
